braindecode/models/sparcnet.py

```diff
@@ -0,0 +1,353 @@
+        Whether to use bias in convolutional layers. Default is True.
+    batch_norm : bool, optional
+        Whether to use batch normalization. Default is True.
+
+    Examples
+    --------
+    >>> x = torch.randn(128, 5, 1000)
+    >>> batch, channels, length = x.shape
+    >>> model = DenseBlock(3, channels, 5, 2)
+    >>> y = model(x)
+    >>> y.shape
+    torch.Size([128, 20, 1000])
+    """
+
+    def __init__(
+        self,
+        num_layers,
+        input_channels,
+        growth_rate,
+        bn_size,
+        drop_rate=0.5,
+        conv_bias=True,
+        batch_norm=True,
+    ):
+        super(DenseBlock, self).__init__()
+        for idx_layer in range(num_layers):
+            layer = DenseLayer(
+                input_channels + idx_layer * growth_rate,
+                growth_rate,
+                bn_size,
+                drop_rate,
+                conv_bias,
+                batch_norm,
+            )
+            self.add_module(f"denselayer{idx_layer + 1}", layer)
+
+
+class TransitionLayer(nn.Sequential):
+    """
+    A pooling transition layer.
+
+    Parameters
+    ----------
+    input_channels : int
+        Number of input channels.
+    output_channels : int
+        Number of output channels.
+    conv_bias : bool, optional
+        Whether to use bias in convolutional layers. Default is True.
+    batch_norm : bool, optional
+        Whether to use batch normalization. Default is True.
+
+    Examples
+    --------
+    >>> x = torch.randn(128, 5, 1000)
+    >>> model = TransitionLayer(5, 18)
+    >>> y = model(x)
+    >>> y.shape
+    torch.Size([128, 18, 500])
+    """
+
+    def __init__(
+        self, input_channels, output_channels, conv_bias=True, batch_norm=True
+    ):
+        super(TransitionLayer, self).__init__()
+        if batch_norm:
+            self.add_module("norm", nn.BatchNorm1d(input_channels))
+        self.add_module("elu", nn.ELU())
+        self.add_module(
+            "conv",
+            nn.Conv1d(
+                input_channels,
+                output_channels,
+                kernel_size=1,
+                stride=1,
+                bias=conv_bias,
+            ),
+        )
+        self.add_module("pool", nn.AvgPool1d(kernel_size=2, stride=2))
+
+
+class SPARCNet(EEGModuleMixin, nn.Module):
+    """Seizures, Periodic and Rhythmic pattern Continuum Neural Network (SPaRCNet) [jing2023]_.
+
+    This is a temporal CNN model for biosignal classification based on the DenseNet
+    architecture.
+
+    The model is based on the unofficial implementation [Code2023]_.
+
+    .. versionadded:: 0.9
+
+    Notes
+    -----
+    This implementation is not guaranteed to be correct, has not been checked
+    by original authors.
+
+    Parameters
+    ----------
+    block_layers : int, optional
+        Number of layers per dense block. Default is 4.
+    growth_rate : int, optional
+        Growth rate of the DenseNet. Default is 16.
+    bn_size : int, optional
```

Comment:
bn sounds like batch_norm on first read. maybe use bottleneck_size to be explicit